eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -669,91 +669,75 @@ explainPlutusFailure ::
-                  ]
-            db = PlutusDebugV1 cm eu scriptbytestring ds pv
-explainPlutusFailure _proxy pv lang scriptbytestring e ds cm eu =
-  -- A script with the wrong number of arguments
-  scriptFail $ PlutusSF line db
-  where
-    ss :: Script era
-    ss = PlutusScript lang scriptbytestring
-    name :: String
-    name = show ss
-    line =
-      pack $
-        unlines
-          ( [ "\nThe plutus script (" ++ name ++ ") fails.",
-              show e,
-              "The protocol version is: " ++ show pv,
-              "It was passed these " ++ show (Prelude.length ds) ++ " data arguments."
+      firstLine = "\nThe " ++ show lang ++ " script (" ++ name ++ ") fails."
+      pvLine = "The protocol version is: " ++ show pv
+      plutusError = "The plutus error is: " ++ show e
+      dataLines =
+        case ds of
+          [dat, redeemer, info] ->
+            case lang of
+              PlutusV1 ->
+                case PV1.fromData info of
+                  Nothing ->
+                    [ "The data is: " ++ show dat,
+                      "The redeemer is: " ++ show redeemer,
+                      "The third data argument, does not translate to a V1 script context\n" ++ show info
+                    ]
+                  Just info2 ->
+                    [ "The data is: " ++ show dat,
+                      "The redeemer is: " ++ show redeemer,
+                      "The script context is:\n" ++ show (pretty (info2 :: PV1.ScriptContext))
+                    ]
+              PlutusV2 ->
+                case PV2.fromData info of
+                  Nothing ->
+                    [ "The data is: " ++ show dat,
+                      "The redeemer is: " ++ show redeemer,
+                      "The third data argument, does not translate to a V2 script context\n" ++ show info
+                    ]
+                  Just info2 ->
+                    [ "The data is: " ++ show dat,
+                      "The redeemer is: " ++ show redeemer,
+                      "The script context is:\n" ++ show (pretty (info2 :: PV2.ScriptContext))
+                    ]
+          [redeemer, info] ->
+            case lang of
+              PlutusV1 ->
+                case PV1.fromData info of
+                  Nothing ->
+                    [ "The redeemer is: " ++ show redeemer,
+                      "The third data argument, does not translate to a V1 script context\n" ++ show info
+                    ]
+                  Just info2 ->
+                    [ "The redeemer is: " ++ show redeemer,
+                      "The script context is:\n" ++ show (pretty (info2 :: PV1.ScriptContext))
+                    ]
+              PlutusV2 ->
+                case PV2.fromData info of
+                  Nothing ->
+                    [ "The redeemer is: " ++ show redeemer,
+                      "The third data argument, does not translate to a V2 script context\n" ++ show info
```

Comment:
Same question as above, wonder if it should say "The second argument"  instead of  third